fenix/app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt

```diff
@@ -168,11 +169,11 @@ class BrowserFragment : BaseBrowserFragment(), UserInteractionHandler {
                 listener = browserToolbarInteractor::onReaderModePressed,
             )
 
-        browserToolbarView.view.addPageAction(readerModeAction)
+        browserToolbarView.view.addPageActionInOrder(readerModeAction, ToolbarActionIdentifier.READER_MODE)
```

Comment:
Is there a way to implement this without using the identifiers by just making sure that the actions added in the correct order? For example, if we know the page action ordering should be:  
    READER_MODE,
    TRANSLATION,
    REVIEW_QUALITY_CHECK,
    SHARE,
    RELOAD,

We would just call addPageAction according to this ordering. The problem will adding these identifier into these internal AC components is that these are very generic actions and AC shouldn't have a concept of what the possible actions nor the ordering it should be. Any browser should be able to just say I have a new page action named "Blah" and be able to add it.

Reply:
Hmm, the thing is `addPageAction` isn't the only one deciding the order. The view only if visible gets added in order in the addPageAction function. However if it is not visible yet, it is just added to the list, without deciding it's ordering. The `invalidateActionActionsWithIdentifier` is the one one which gets called when a view becomes visible and hence puts the view at the right place.

Reply:
@rocketsroger Could you take a look here and see if there is another path forward? I think we need to try to have a way where we are not including https://github.com/mozilla-mobile/firefox-android/pull/6025/files#diff-f854bc196fb764d678d6f74a529fd328a64052ff2c57bcfeeb8d480c8ba63c97R32 in the AC internals because not every browser is expected to have these page actions nor this particular ordering.

Reply:
I am going through another route to include a parameter called `weight`in the `Toolbar.Action` class. The concept behind that will be a weight will be passed every time a `pageAction` is created (which is on the consumer/fenix side). The left one will have the smallest weight and the right one the most. These will be positive integers and there will be a default weight of -1 which wouldn't have effect anything, i.e it will be added the way it is added today.

Reply:
I've updated the PR with above concept.